receiver/otlpreceiver/otlp.go

```diff
@@ -180,6 +187,11 @@ func (r *otlpReceiver) Shutdown(ctx context.Context) error {
 		if r.serverGRPC != nil {
 			r.serverGRPC.GracefulStop()
 		}
+
+		r.shutdownWG.Wait()
+
+		// delete the receiver from the map.
+		delete(receivers, r.cfg)
```

Comment:
Please elaborate a bit more why :) Since we spent some time understanding this issue, would be good for others to read this and know why this is happening.

Reply:
Done.